packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -3463,6 +3469,30 @@ export class ContainerRuntime
 		return result;
 	}
 
+	detachHead(): { merge(): void; pause(): Promise<void>; dispose(): void } {
+		const checkpoint = this.outbox.getBatchCheckpoints(true);
+		let paused = false;
+		const branchInfo = {
+			dispose: () => {
+				checkpoint.mainBatch.rollback();
+				checkpoint.unblockFlush();
+			},
+			pause: async () => {
+				if (!paused) {
+					paused = true;
+					return this._deltaManager.inbound.pause();
+				}
+			},
+			merge: () => {
```

Comment:
i also don't really like `merge` here, maybe `commit`, `persist`, `send`, etc?

Reply:
How about `commitChanges` and `discardChanges` instead of `merge` and `dispose`